assets/src/edit-story/components/library/panes/text/textSets/getTextSetReducer.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const TEXT_SET_ACTIONS = {
+  RESET: 'reset',
+  RENDER_NEXT_TEXT_SET: 'render_next_text_set',
+};
+
+export function getTextSetReducer(textSets, allTextSets, size = 10) {
```

Comment:
I can update this in the next filter PR. Good idea.

Reply:
Oh nevermind, we are using `useReducer`.

Reply:
```
const [{ filteredTextSets, renderedTextSets }, dispatch] = useReducer(
    textSetReducer,
    {
      filteredTextSets: [],
      renderedTextSets: [],
    }
  );
```

Reply:
As for the unloading of DOM, I agree, I think we are going to have to rethink how we do render preview components as this will only become a bigger problem with layout sets as well.